File: Algorithms/Pattern_Searching_Algorithm/C++/naivePatternSearch.cpp
#include <bits/stdc++.h> 
using namespace std; 

void patSearchinng(string &txt,string &pat){
    int m=pat.length();
    int n=txt.length();
    for(int i=0;i<=(n-m);i++){
        int j;
        for(j=0;j<m;j++) {
            if(pat[j]!=txt[i+j])
                break;
        }
        if(j==m)
        cout<<i<<" ";
    }
}
 
int main() 
{ 
    string txt = "ABCABCD";string pat="ABCD";
    cout<<"All index numbers where pattern found:"<<" ";
    patSearchinng(txt,pat);
    
    return 0; 
} 

File: Algorithms/Pattern_Searching_Algorithm/Java/naivePatternSearch.java
import java.util.*;
import java.io.*;
import java.lang.*;
  
class GFG { 
       
    static void patSearchinng(String txt,String pat){
    int m=pat.length();
    int n=txt.length();
    for(int i=0;i<=(n-m);i++){
        int j;
        for(j=0;j<m;j++)
        if(pat.charAt(j)!=txt.charAt(i+j))
        break;
        
        if(j==m)
        System.out.print(i+" ");
    }
    }
  
    public static void main(String args[]) 
    {   String txt = "ABCABCD";String pat="ABCD";
        System.out.print("All index numbers where pattern found: ");
        patSearchinng(txt,pat);  
    } 
} 